### DISCUSSION TITLE
dificulties building environnement on windows

### DISCUSSION THREAD

**Discussion by Gadzbee** [OpenRadioss/OpenRadioss#2469] [Q&A]
Hello everyone,

I'm currently following the youtube tutorial in order to install open radioss and all of the softwares needed but when i'm in the command shell section and entering 

C:\Work\SOURCE\OpenRadioss-latest-20240329\starter>build_windows.bat -arch = win64

The command failed in the 2055th step with the error message 

C:\Work\SOURCE\OpenRadioss-latest-20240329\common_source\comm\radioss_set_env_variable.c(36,10): fatal error: 'sys/resource.h' file not found
   36 | #include <sys/resource.h>
      |          ^~~~~~~~~~~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.

Errors in build encontered !!!

Thanks in advance ! 

Aurélien

**Comment by OlivierWienholtz** [comment]
Hello Aurelien

What compilers have you installed ? 
How is your build environment configured ? 
Indeed, in this file there is a section with protected by pre-processor directives.
The used directive is _WIN64. It is defined in INTEL & Visual Studio CL compiler.

Here the Linux par was loaded & not the Windows one. This is strange.

It could help if you can log more infos on the Cmake build. There are some interesting informations in the Headers.

Hope it helps, 
Olivier W.

**Comment by Gadzbee** [comment]
Hello Olivier,

I have installed Visual studio 2022 and IntelOne API. For the environnement build i followed the tutorial by Marian step by step. I don't know if this is because i'm using my computer from the laborartory. And do you know by any chance how the header file of the Cmake build is named so i can log the info ? 

I'm really sorry i'm not fully understanding how this works for now but your help is very precious.
Thanks again

Aurélien

**Comment by OlivierWienholtz** [comment]
Hello Aurelien

One Question : what version of OneAPI did you install ? 
I just saw that latest OneAPI was released 10 days  ago.
Did you install this one ? 

can you type in the shell : ifx --version ?

I'll do some checks on my side.

Many Thanks,
Best Regard,
Olivier

**Comment by Gadzbee** [comment]
Hello Olivier,

the shell is sending me this message : 

Intel(R) Fortran Compiler for applications running on IA-32, Version 2024.1.0 Build 20240308
Copyright (C) 1985-2024 Intel Corporation. All rights reserved.

ifx: command line warning #10006: ignoring unknown option '/-version'
ifx: command line warning #10161: unrecognized source type '.'; object file assumed
ifx: command line warning #10161: unrecognized source type '..'; object file assumed
Microsoft (R) Incremental Linker Version 14.39.33523.0
Copyright (C) Microsoft Corporation.  All rights reserved.

-out:..exe
-subsystem:console
.
..
LINK : fatal error LNK1104: impossible d'ouvrir le fichier '..exe'

Thanks again for your help

Best regard,
Aurélien

**Comment by OlivierWienholtz** [comment]
Hello Gadzbee

Now I understand what happened : 
You have loaded the 32 bit version of the compiler & not the 64 bit.
OpenRadioss must be build with 64 bit Compiler.
Therefore the C builds failed .

Please check if you have installed the 64 bit version on your machine 
& Load the 64 bit compiler. 
If you have both calling the compiler 
call "c:\Program Files (x86)\Intel\oneAPI\setvars.bat" intel64 

should d the trick

Best Regards,
Olivier W.

**Comment by Gadzbee** [comment]
Hello Oliver,

Your response helped me a lot, I used the windows64 version of intel. But now i seems to have a new problem with the macro called ASSERT. I tried to look at the corresponding file but still not understanding what is wrong : 

[2/9184] Building Fortran preprocessed starter\CMakeFiles\starter_.dir\source\restart\ddsplit\inter_tools.F-pp.f
FAILED: starter/CMakeFiles/starter_.dir/source/restart/ddsplit/inter_tools.F-pp.f starter/CMakeFiles/starter_.dir/source/restart/ddsplit/inter_tools.F.obj.ddi
C:\WINDOWS\system32\cmd.exe /C "C:\PROGRA~2\Intel\oneAPI\compiler\latest\bin\ifx.exe -fpp  -IC:\Work\SOURCE\OpenRadioss-latest-20240411\starter\..\common_source\includes -IC:\Work\SOURCE\OpenRadioss-latest-20240411\starter\share\c_inc -IC:\Work\SOURCE\OpenRadioss-latest-20240411\starter\share\includes -IC:\Work\SOURCE\OpenRadioss-latest-20240411\starter\share\r8 -IC:\Work\SOURCE\OpenRadioss-latest-20240411\starter\share\spe_inc -IC:\Work\SOURCE\OpenRadioss-latest-20240411\cbuild__ninja\starter\CMakeFiles\includes_starter_ /W1 /nologo /fpp /libs:dll /threads /O2 /DNDEBUG -module:starter -P C:\Work\SOURCE\OpenRadioss-latest-20240411\starter\source\restart\ddsplit\inter_tools.F -Fistarter\CMakeFiles\starter_.dir\source\restart\ddsplit\inter_tools.F-pp.f && "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -E cmake_ninja_depends --tdi=starter\CMakeFiles\starter_.dir\FortranDependInfo.json --lang=Fortran --src=starter\CMakeFiles\starter_.dir\source\restart\ddsplit\inter_tools.F-pp.f --out=starter\CMakeFiles\starter_.dir\source\restart\ddsplit\inter_tools.F-pp.f --dep=starter\CMakeFiles\starter_.dir\source\restart\ddsplit\inter_tools.F-pp.f.d --obj=starter\CMakeFiles\starter_.dir\source\restart\ddsplit\inter_tools.F.obj --ddi=starter\CMakeFiles\starter_.dir\source\restart\ddsplit\inter_tools.F.obj.ddi"

inter_tools.F(2411): #error: incomplete macro call ASSERT.
ninja: build stopped: subcommand failed.

Terminating

Best regards,

Aurélien

**Comment by Gadzbee** [comment]
I uninstalled and reinstalled all the software and it finally worked. Thank you so much Olivier for your follow-up!

**Comment by shishirsanjeeva** [comment]
Hi Guys
I encountered an issue while building in .../starter
"FAILED: starter_win64.exe"
fatal error LNK1112: module machine type 'x86' conflicts with target machine type 'x64'
![Screenshot 2024-05-04 203749](https://github.com/OpenRadioss/OpenRadioss/assets/67865221/6426d9c2-06c9-4c23-858b-e0c5f6780cff)

**Reply by OlivierWienholtz** [reply]
Hello,

OpenRadioss is a 64 bit executable.
To build it, you need 64 Bit Intel Compilers. 

Pls check Ifort or icx compiler version:

ifort --version
icx -version

**Reply by shishirsanjeeva** [reply]
Hi Olivier
I checked the versions and both of them are 64 bit.
![image](https://github.com/OpenRadioss/OpenRadioss/assets/67865221/0b528118-ed0c-4631-9b28-43582e4db89a)
